packages/dds/CMakeLists.txt

```diff
@@ -48,6 +38,162 @@ set(SRC
+find_package(fastrtps QUIET)
+
+if(fastrtps_DIR)
+    message(STATUS "Found a fastrtps")
+    target_link_libraries(${PROJECT_NAME}
+        PRIVATE
+            soss::core
+            fastrtps
+            $<$<PLATFORM_ID:Windows>:yaml-cpp>
+    )
+    if (SOSS-DDS_BUILD_TESTS)
+        target_link_libraries(${PROJECT_NAME}_test_conversion
+            PRIVATE
+                fastrtps
+                soss::core
+        )
+    endif()
+
+else()
+    message(STATUS "Using the fastrtps as external project")
+    # Fast-RTPS needs foonathan too
+    unset(foonathan_memory_DIR CACHE) # Reviewer! Do we need to clear CACHE? Copied from uXRCE-DDS-Agent
+    find_package(foonathan_memory QUIET)
+    if(NOT foonathan_memory_FOUND)
+        ExternalProject_Add(foonathan_memory
+            GIT_REPOSITORY
+                https://github.com/foonathan/memory.git
+            GIT_TAG
+                master # ${_foonathan_memory_tag} (set(_foonathan_memory_tag master))
+            PREFIX
+                ${PROJECT_BINARY_DIR}/foonathan_memory
+            INSTALL_DIR
+                ${PROJECT_BINARY_DIR}/temp_install/foonathan_memory
+            CMAKE_CACHE_ARGS
+                -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+                -DFOONATHAN_MEMORY_BUILD_EXAMPLES:BOOL=OFF
+                -DFOONATHAN_MEMORY_BUILD_TESTS:BOOL=OFF
+                -DFOONATHAN_MEMORY_BUILD_TOOLS:BOOL=ON
+                -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON
+                -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+            )
+    endif()
+
+    # Fast RTPS.
+    unset(fastrtps_DIR CACHE) # Reviewer! Do we need to clear CACHE? Copied from uXRCE-DDS-Agent
+    find_package(fastrtps QUIET)
+    #find_package(fastrtps ${_fastrtps_version} EXACT QUIET)
+    if(NOT fastrtps_FOUND)
+        ExternalProject_Add(fastrtps
+            GIT_REPOSITORY
+                https://github.com/eProsima/Fast-RTPS.git
+            GIT_TAG
+                v1.9.3 #${_fastrtps_tag}
+            PREFIX
+                ${PROJECT_BINARY_DIR}/fastrtps
+            INSTALL_DIR
+                ${PROJECT_BINARY_DIR}/temp_install/fastrtps
+            CMAKE_CACHE_ARGS
+                -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+                -DCMAKE_PREFIX_PATH:PATH=${CMAKE_PREFIX_PATH};${PROJECT_BINARY_DIR}/temp_install
+                -DBUILD_SHARED_LIBS:BOOL=ON
+                -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
+                -DCMAKE_TOOLCHAIN_FILE:PATH=${CMAKE_TOOLCHAIN_FILE}
+                -DTHIRDPARTY:BOOL=ON
+                -DSECURITY:BOOL=OFF #${SOSS-DDS_SECURITY_PROFILE}
+            DEPENDS
+                foonathan_memory
+            )
+
+        link_directories(${PROJECT_BINARY_DIR}/temp_install/fastrtps/lib)
+        include_directories(${PROJECT_BINARY_DIR}/temp_install/fastrtps/include)
+        add_library(libfastrtps IMPORTED SHARED GLOBAL)
+        add_library(libfastcdr IMPORTED SHARED GLOBAL)
+        add_dependencies(libfastrtps fastrtps)
```

Comment:
Analogous line for `fastcdr`?

Reply:
`fastcdr` isn't a known symbol at this point, we only need to load it as library within `libfastcdr` by locating the `.so` file.
It would be need if we add `fastcdr` as an external project instead of building `fastrtps` with *THIRDPARTY*.

Reply:
talked in person.